src/libcollections/string.rs

```diff
@@ -1015,8 +1015,7 @@ impl String {
     }
 
     /// Create a draining iterator that removes the specified range in the string
-    /// and yields the removed chars from start to end. The element range is
-    /// removed even if the iterator is not consumed until the end.
```

Comment:
The information about removal is good I think. You might otherwise think that only elements extracted from the iterator are removed from the String or Vec.